File: 1878-check-if-array-is-sorted-and-rotated/1878-check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        vector<int> arr = nums;
        arr.insert(arr.end(), arr.begin(), arr.end());
        
        for(int i=0; i<nums.size(); i++) {
            bool isSorted = true;
            for(int j=i+1; j<i+nums.size(); j++) {
                if(arr[j] < arr[j-1]) {
                    isSorted = false;
                    break;
                }
            }
            if(isSorted) return true;
        }

       
        return false;
    }
};

File: 1878-check-if-array-is-sorted-and-rotated/1878-check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int n = nums.length;
        int[] arr = new int[2*n];
        int m = arr.length;
        System.arraycopy(nums, 0, arr, 0, n);
        System.arraycopy(nums, 0, arr, n, n);
        int count = 1;
        if(n==1) return true;
        for(int i=1; i<m; i++) {
            if(arr[i-1] <= arr[i]) {
                count++;
                if(count == nums.length) return true;
            }
            else if(count == nums.length) return true;
            else count = 1;
        }

        return false;
    }
}